compiler/src/dotty/tools/backend/sjs/JSEncoding.scala

```diff
@@ -92,6 +92,16 @@ object JSEncoding {
         returnLabelName = Some(freshName("_return"))
       js.Ident(returnLabelName.get)
     }
+
+    def withNewReturnableScope(tpe: jstpe.Type)(body: => js.Tree)(implicit pos: ir.Position): js.Tree = {
```

Comment:
It seems `body` doesn't need to be by-name

Reply:
It is needed. `returnLabelName` will be mutated from `None` to `Some(...)` somewhere inside the execution of `body` (or not if there is no `Return()` node inside the body).

Reply:
I may be missing something. The value of `returnLabelName` is not saved (or used) before the evaluation of `body`. So what difference does it make to evaluate it before the function is called?

Reply:
Oh, right. I didn't think about it like that. Yeah I guess it can be made a by-value parameter, then, but I'd advocate for a different name in that case, for example `wrapInReturnLabeledIfRequired` or something like that. The name `withXyz` suggests that something by-name is happening, at least to me.

Reply:
Good catch - the Scala 2 version of this used ScopedVars in a way that required by-name, but as a mutable handle to the same `localNames`, the Dotty version has no such need. Made this by-value and updated the method's name accordingly.